modules/extensions-spring-support/src/main/java/org/mule/module/extension/internal/capability/xml/schema/builder/SchemaBuilder.java

```diff
@@ -197,151 +175,45 @@ private SchemaBuilder importMuleExtensionNamespace()
-
-        addConnectionProviderElement(choice, configurationModel);
-        addDynamicConfigPolicyElement(choice, configurationModel);
-        registerParameters(config, choice, configurationModel.getParameterModels());
-        config.setAnnotation(createDocAnnotation(configurationModel.getDescription()));
-
+        new ConfigurationSchemaDelegate(this).registerConfigElement(schema, configurationModel);
         return this;
     }
 
-    private Attribute createNameAttribute()
+    Attribute createNameAttribute()
     {
         return createAttribute(SchemaConstants.ATTRIBUTE_NAME_NAME, DataType.of(String.class), true, NOT_SUPPORTED);
     }
 
     public SchemaBuilder registerOperation(OperationModel operationModel)
     {
-        String typeName = capitalize(operationModel.getName()) + TYPE_SUFFIX;
-        registerProcessorElement(operationModel, typeName);
-        registerOperationType(typeName, operationModel);
-
+        new OperationSchemaDelegate(this).registerOperation(schema, operationModel);
         return this;
     }
 
     public SchemaBuilder registerMessageSource(SourceModel sourceModel)
     {
-        String typeName = capitalize(sourceModel.getName()) + TYPE_SUFFIX;
-        registerSourceElement(sourceModel, typeName);
-        registerSourceType(typeName, sourceModel);
 
+        new SourceSchemaDelegate(this).registerMessageSource(schema, sourceModel);
         return this;
     }
 
-    private void addConnectionProviderElement(ExplicitGroup all, ConfigurationModel configurationModel)
-    {
-        ExtensionModel extensionModel = configurationModel.getExtensionModel();
-        if (!getConnectedOperations(extensionModel).isEmpty())
-        {
-            TopLevelElement objectElement = new TopLevelElement();
-
-            objectElement.setMinOccurs(getFirstImplicit(extensionModel.getConnectionProviders()) != null
-                                       ? ZERO
-                                       : ONE);
-
-            objectElement.setMaxOccurs("1");
-            objectElement.setRef(MULE_EXTENSION_CONNECTION_PROVIDER_ELEMENT);
-
-            all.getParticle().add(objectFactory.createElement(objectElement));
-        }
-    }
-
-    private void addDynamicConfigPolicyElement(ExplicitGroup all, ConfigurationModel configurationModel)
-    {
-        if (!getDynamicParameters(configurationModel).isEmpty())
-        {
-            TopLevelElement objectElement = new TopLevelElement();
-            objectElement.setMinOccurs(ZERO);
-            objectElement.setMaxOccurs("1");
-            objectElement.setRef(MULE_EXTENSION_DYNAMIC_CONFIG_POLICY_ELEMENT);
 
-            all.getParticle().add(objectFactory.createElement(objectElement));
-        }
-    }
-
-    private void registerParameters(ExtensionType type, ExplicitGroup sequence, Collection<ParameterModel> parameterModels)
+    void registerParameters(ExtensionType type, ExplicitGroup choice, Collection<ParameterModel> parameterModels)
```

Comment:
Delegates were for registration of components: Configuration, ConnectionProvider, Source and Operations. Since registration of parameters is cross-component, then the implementation remains in the builder. Also, the implementation is far more simple than the one the components use, so the need of extracting this to a different class is not so strong.